controllers/upgrade.go

```diff
@@ -144,6 +164,9 @@ func (r *RollingUpgradeReconciler) SelectTargets(rollingUpgrade *v1alpha1.Rollin
 	if rollingUpgrade.UpdateStrategyType() == v1alpha1.RandomUpdateStrategy {
 		for _, instance := range scalingGroup.Instances {
 			if r.IsInstanceDrifted(rollingUpgrade, instance) {
+				if common.ContainsEqualFold(v1alpha1.ASGTerminatingInstanceStates, aws.StringValue(instance.LifecycleState)) {
```

Comment:
Should we include this if condition within `IsInstanceDrifted`?

Reply:
Haha! I had the same thoughts too. I see no harm in doing so. I think I can move it there.